cloudimg/aws.py

```diff
@@ -40,18 +57,22 @@ class AWSPublishingMetadata(PublishingMetadata):
                                            interfaces. By default this option
                                            is enabled.
         billing_products (list, optional): Billing product identifiers
+
+        boot_mode (str, optional): The boot mode for booting up the AMI on EC2.
     """
 
     def __init__(self, *args, **kwargs):
         self.ena_support = kwargs.pop('ena_support', True)
         self.sriov_net_support = kwargs.pop('sriov_net_support', 'simple')
         self.billing_products = kwargs.pop('billing_products', None)
+        self.boot_mode = AWSBootMode(kwargs.pop('boot_mode', None))
```

Comment:
will this work?
```
>>> class AWSBootMode(enum.Enum):
...     uefi = "uefi"
...     bios = "legacy-bios"
...     hybrid = "uefi-preferred"
...     not_set = None
... 
>>> # from push source
>>> class BootMode(enum.Enum):
...     hybrid = "hybrid"
...     uefi = "uefi"
...     legacy = "legacy"
... 
>>> AWSBootMode("hybrid")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/enum.py", line 385, in __call__
    return cls.__new__(cls, value)
  File "/usr/lib64/python3.10/enum.py", line 710, in __new__
    raise ve_exc
ValueError: 'hybrid' is not a valid AWSBootMode
>>> AWSBootMode(BootMode.hybrid)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/enum.py", line 385, in __call__
    return cls.__new__(cls, value)
  File "/usr/lib64/python3.10/enum.py", line 710, in __new__
    raise ve_exc
ValueError: <BootMode.hybrid: 'hybrid'> is not a valid AWSBootMode
```
it works with value what is expected by boto3 though
```
>>> AWSBootMode("uefi-preferred")
<AWSBootMode.hybrid: 'uefi-preferred'>
```

Reply:
We'll map the Enum values from `pushsource` to the proper values from boto3 in our pubtool hence this one has just the boto3 values.

Reply:
@lslebodn you're correct in the assumption we can't pass the `None` value to `BootMode`.

It fails as shown below:

```
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter BootMode, value: None, type: <class 'NoneType'>, valid types: <class 'str'>
```

I've patched it [here](https://github.com/release-engineering/cloudimg/compare/c494d061f43f5e8ee8c41b1ba3c7e80f34ea0cfb..7132a0955c66c571dad38eb276231216c8bc4d71), PTAL again

It's now passing the manual test.

I've added a new automated test for the `None` case as well